numba/tests/test_event.py

```diff
@@ -117,6 +141,48 @@ def foo():
             lifted_cres.metadata["timers"]["compiler_lock"],
             float,
         )
+        self.assertIsInstance(
+            lifted_cres.metadata["timers"]["llvm_lock"],
+            float,
+        )
+
+    def test_timing_property(self):
+        a = tuple(string.ascii_lowercase)
+
+        @njit
+        def bar(x):
+            acc = 0
+            for i in literal_unroll(a):
+                if i in {'1': x}:
+                    acc += 1
+                else:
+                    acc += np.sqrt(x[0, 0])
+            return np.sin(x), acc
+
+        @njit
+        def foo(x):
+            return bar(np.zeros((x, x)))
+
+        foo(1)
+
+        def get_timers(fn):
+            md = fn.get_metadata(fn.signatures[0])
+            return md['timers']
+
+        foo_timers = get_timers(foo)
+        bar_timers = get_timers(bar)
+
+        # Check: time spent in bar() must be longer than in foo()
+        self.assertLess(bar_timers['llvm_lock'],
+                        foo_timers['llvm_lock'])
+        self.assertLess(bar_timers['compiler_lock'],
+                        foo_timers['compiler_lock'])
+
+        # Check: time spent in LLVM lock must be less than in compiler
+        self.assertLess(foo_timers['llvm_lock'],
+                        foo_timers['compiler_lock'])
+        self.assertLess(bar_timers['llvm_lock'],
+                        bar_timers['compiler_lock'])
```

Comment:
Suggest updating as:
```patch
diff --git a/numba/tests/test_event.py b/numba/tests/test_event.py
index 264807d0e..ed24741de 100644
--- a/numba/tests/test_event.py
+++ b/numba/tests/test_event.py
@@ -5,7 +5,7 @@ import numpy as np
 
 from numba import njit, jit, literal_unroll
 from numba.core import event as ev
-from numba.tests.support import TestCase
+from numba.tests.support import TestCase, override_config
 
 
 class TestEvent(TestCase):
@@ -163,14 +163,17 @@ class TestEvent(TestCase):
         def foo(x):
             return bar(np.zeros((x, x)))
 
-        foo(1)
+        with override_config('LLVM_PASS_TIMINGS', True):
+            foo(1)
 
-        def get_timers(fn):
+        def get_timers(fn, prop):
             md = fn.get_metadata(fn.signatures[0])
-            return md['timers']
+            return md[prop]
 
-        foo_timers = get_timers(foo)
-        bar_timers = get_timers(bar)
+        foo_timers = get_timers(foo, 'timers')
+        bar_timers = get_timers(bar, 'timers')
+        foo_llvm_timer = get_timers(foo, 'llvm_pass_timings')
+        bar_llvm_timer = get_timers(bar, 'llvm_pass_timings')
 
         # Check: time spent in bar() must be longer than in foo()
         self.assertLess(bar_timers['llvm_lock'],
@@ -184,6 +187,12 @@ class TestEvent(TestCase):
         self.assertLess(bar_timers['llvm_lock'],
                         bar_timers['compiler_lock'])
 
+        # Check: time spent in LLVM itself must be less than in the LLVM lock
+        self.assertLess(foo_llvm_timer.get_total_time(),
+                        foo_timers['llvm_lock'])
+        self.assertLess(bar_llvm_timer.get_total_time(),
+                        bar_timers['llvm_lock'])
+
```
what do you think? The original problem in https://github.com/numba/numba/pull/6636#issuecomment-760168944 would not be caught by what's present at the moment?

Reply:
Makes sense, testing the LLVM time vs LLVM lock

Reply:
applied suggestion in https://github.com/numba/numba/pull/6636/commits/c34bf609fbdae6bedf43e468387224a21d2928fe

Reply:
thanks